packages/babel-types/src/definitions/core.ts

```diff
@@ -1964,15 +1965,29 @@ defineType("TemplateElement", {
   builder: ["value", "tail"],
   fields: {
     value: {
-      validate: assertShape({
-        raw: {
-          validate: assertValueType("string"),
-        },
-        cooked: {
-          validate: assertValueType("string"),
-          optional: true,
+      validate: chain(
+        assertShape({
+          raw: {
+            validate: assertValueType("string"),
+          },
+          cooked: {
+            validate: assertValueType("string"),
+            optional: true,
+          },
+        }),
+        function templateElementCookedValidator(node: t.TemplateElement) {
+          const raw = node.value.raw;
+          if (node.value.cooked != null) {
+            return;
+          }
+
+          let cooked = null;
+          try {
+            cooked = unraw(raw);
```

Comment:
Q: should we throw if user provided inconsistent `cooked`? Currently we mute such errors and trust `raw`, but in this case either the `raw` or `cooked` could be incorrect.

Reply:
> should we throw if user provided inconsistent `cooked`?

I'm not sure.

> Currently we mute such errors and trust `raw`, but in this case either the `raw` or `cooked` could be incorrect.

Can you give an example?
I test that fn `\u` is legal and `\u` is not.
Looks like it should be checked in `TemplateLiteral`.
Then we can do it later.😂

Reply:
For example, when user provided both `raw` and `cooked`:
```js
t.templateElement({
  raw: "face",
  cooked: "faec" // which one is correct? face or faec?
})
```

The current main behaviour is leave it as-is. The PR behaviour is to ignore `cooked` and reassign `cooked` to be `raw`. Now thanks to `unraw` we know that this is not a valid template element. Should we throw a validation error in this case?

Reply:
Make this a behavior in babel8?
But I don't know if jest snap supports this.

Reply:
Now we ignore the `cooked` passed in by the user.
As for `raw` exceptions I think it can be done in the future.